jaeger-core/src/test/java/com/uber/jaeger/reporters/RemoteReporterTest.java

```diff
@@ -103,4 +109,97 @@ public void testRemoteReporterFlushTimerThread() throws Exception {
     }
     assertFalse(flushTimerThreadCount == 0);
   }
+
+  // Starts a number of threads. Each can fill the queue on its own, so they will exceed its
+  // capacity many times over
+  @Test
+  public void testReportDoesntThrowWhenQueueFull() throws Exception {
+    final AtomicBoolean exceptionWasThrown = new AtomicBoolean(false);
+
+    int threadsCount = 10;
+    final CyclicBarrier barrier = new CyclicBarrier(threadsCount);
+    List<Thread> threads = new ArrayList<>();
+    for (int i = 0; i < threadsCount; i++) {
+      Thread t = createSpanReportingThread(exceptionWasThrown, barrier);
+      threads.add(t);
+      t.start();
+    }
+
+    for (Thread t : threads) {
+      t.join();
+    }
+
+    assertFalse(exceptionWasThrown.get());
+  }
+
+  private Thread createSpanReportingThread(final AtomicBoolean exceptionWasThrown,
+      final CyclicBarrier barrier) {
+    return new Thread(new Runnable() {
+          @Override
+          public void run() {
+            for (int x = 0; x < maxQueueSize; x++) {
+              try {
+                barrier.await();
+                reporter.report(newSpan());
+              } catch (Throwable e) {
+                e.printStackTrace();
+                exceptionWasThrown.set(true);
+              }
+            }
+          }
+        });
+  }
+
+  @Test
+  public void testAppendWhenQueueFull() {
+    // change sender to blocking mode
+    sender.permitAppend(0);
+
+    for (int i = 0; i < maxQueueSize; i++) {
+      reporter.report(newSpan());
+    }
+
+    // When: at this point the queue is full or there is one slot empty (if the worker thread has
+    // already picked up some command). We add two spans to make sure that we overfill the queue
```

Comment:
if the workers are still reading from the queue, then it looks like you have a race condition and no guarantee that the next two commands would actually cause a queue overflow